### ISSUE TITLE
Precipitation exponential factor: available for all precipitation parameterisations

### ISSUE THREAD

**Issue by flocolleoni** [pism/pism#447]
Hello,

I have some questions about the way to correct precipitation with temperature changes and elevation changes in PISM 1.1

I see that one can prescribe the atmospheric precip exponential factor (clausius-Clapeyron) to correct precip with temperature changes only in paleo climatic routines Paleoprecip or searise Greenland. 
For example, this implies that if projecting to 2300 and using a precip method that is not paleo, e.g. directly prescribing precip from climate projections in which ice sheet geometry has not been changed throughout the simulation, there is no possibilty to readjust precipitation with temperature changes due to elevation changes (which are large in the far future) in pism. 
Wouldn't it make sense to implement this precipitation factor for all parametrisations of precipitation?

On another hand, there is the "precip_lapse_rate" in m/year/km, which is a highly non intuitive variable. What was the purpose of this variable?

best regards

**Comment by aaschwanden** [comment]
Hi Florence,

I’m currently traveling without a computer so I can’t dig into details.

“Paleoprecpi” is clearly a misleading name, and changing it has been on my to do list for a long time, I just keep forgetting.

Nonetheless, the modifier works in any context where you read in a precipitation field, including projections. That is, you can make it “wetter” as a function of temperature increase. This is, for example, what we’ve used in a manuscript last year (Aschwanden et al, 2019).

The precip_lapse_rate is the precipitation equivalent of a temperature lapse rate, it let’s you simulate an increase (decrease) in precip with increasing elevation. I’ve never used it and would be very careful to use it. It may be useful for synthetic setups though. 

@ckhroulev: should we change it before we release the next stable version?

**Comment by flocolleoni** [comment]
Hi Andy,

Thanks for your answer!

What I mean is that:

Is there a simple way of having precip changes when prescribing a steady
state climate for example, without using paleoprecip scheme? Temperature
lapse rate can be prescribed but precip exponential factor is only
accounted for in paleoprecip or searise Greenland routine. But in the case
one prescribes steady state climate conditions via -atmosphere given, is
there a way of changing precip through the exponential factor? It is not
clear to me from the doc and from the routines.

Flo

Le dim. 12 janv. 2020 à 10:03, Andy Aschwanden <notifications@github.com> a
écrit :

**Comment by aaschwanden** [comment]
Flo,

The paleoprecip modifier should work atmosphere given, I used it all the time. Have you tried it? If it’s not working then it’s a bug.

**Comment by flocolleoni** [comment]
Sorry I think I was not clear.

yes paleoprecip works with -atmosphere,given.

What I mean is, in the case you don't use paleoprecip and you prescribe constant climate forcing. If you prescribe a temperature lapse rate, then if your elevation changes, temperature is then adjusted via the lapse rate.

This does not seem to be the case for precipitation. Because the precip exponantial factor is only red in paleoprecip routine or searise greenland routine. This implies that when you prescribe a constant climate forcing, the precip exponantial factor is not accounted for, only the temperature laspe rate is. This exponantial factor is activated only if you use then paleoprecip or searise Greenland in addition to -atmosphere given or other atmosphere parameterisation. But I don't want to use paleoprecip or searise greenladn. I just would like to run a simple equilibrium simulation, with constant climate forcing and no additional prescribed delta_T or delta_P. In this case, I see that PISM does not apply any precipitation corrections of elevation changes (you can only do this if you prescribed this strange scalar precip_lapse_rate parameter, but not via the precip exponantial factor).

In other words, in the LapseRates.cc routine, the precip exponantial factor is not red, only this "precip_lapse_rate" in (m/yr/km). Would it be possible to also correct precipitation with this precip-exponantial_factor (that is set in the pism_config.cdl) for any of the atmosphere model, without activating paleoprecip or searise Greenalnd?

I don't know if it is more clear now.

Flo

**Comment by aaschwanden** [comment]
Ah I think I now understand.

I think that’s possible to do and most likely not too hard. While I won’t be back in my office for another 4 weeks, maybe Constantine will be able to look into this.

**Comment by talbrecht** [comment]
Hej Florence, it was me who introduced the precip scaling with surface elevation change (#416). However, it has not yet been merged into the dev branch?! You can find the motivation (also for the unit) in https://doi.org/10.5194/tc-2019-71, soon to be published. Other models often calculate a local temperature based on external temperature forcing and lapse-rate adjusted and then adjust the precipitation accordingly. In PISM the same can be achieved, but in two steps. And in terms of a modifier (`lapse`) with additonal option `-precip_lapse_scaling` it should work with every atmophere model.

**Comment by ckhroulev** [comment]
Florence,

It looks like you would like PISM to adjust (scale) precipitation in response to changes in air temperature *that are computed by the model* (as opposed to *provided by the user* as with `paleo_precip`).

In particular, if ice geometry changes and this change affects air temperature via an elevation lapse rate, the resulting change in air temperature should be able to influence precipitation by way of an exponential factor.

Is this understanding correct?

**Comment by ckhroulev** [comment]
@talbrecht #416 was not (yet) merged into the `dev` branch for two reasons.

- I have to focus on high-priority tasks.
- It confuses matters: I see that adding this feature to the lapse rate code may be the easiest way to achieve the goal in the short term.

   However, I think it should be implemented elsewhere to make the code easier to document, use, understand, and maintain.

**Comment by flocolleoni** [comment]
Dear Constantine,

Yes it is exactly that.

Then in the particular case of using paleoprecip, both temperature delta_T
and temperature correction due to elevation changes would be accounted for
when correcting precipitation.

**Comment by talbrecht** [comment]
Dear @ckhroulev , that wasn't meant as a reproach, I fully understand the more urgent tasks. And yes, there are certainly more clever ways to implement and document this code. I am always a bit pragmatic at first, but I am open for discussions and iterations.

**Comment by ckhroulev** [comment]
Hmm.

Here's one idea: we could add a new modifier that reads a reference air temperature field from a file, then (during the run) computes `dT = T_reference - T_modeled` and uses this temperature difference to scale precipitation (if requested).

This mechanism would be pretty generic because it would not be tied to a particular air temperature model. It could respond to both `delta_T` (scalar temperature offsets read from a file) and temperature changes corresponding an elevation lapse rate. However, it seems to me that in many cases it would not be clear how to compute `T_reference` and so this mechanism would not be as generic as it seems. It would work fine in constant-climate setups, though.

Here's a second idea (currently implemented in the `dev` branch):

1. Keep `paleo_precip` (but rename it to something like `precip_scaling`).
2. Implement a version of #416.

I renamed `-atmosphere lapse_rate` to `-atmosphere elevation_change` to reflect the fact that it implements some adjustments which are functions of the change of surface elevation but are *not* lapse rates. (The confusion resulting from what seems to be an abuse of the term ["lapse rate"](https://en.wikipedia.org/wiki/Lapse_rate) is the main reason I did not merge #416 sooner.)

In this implementation `-atmosphere elevation_change` adds scaling of precipitation in response to the temperature change due to the elevation lapse rate for temperature. This code does the right thing when combined with `precip_scaling` (formerly known as `paleo_precip`), i.e. we can capture the influence on precipitation of both elevation changes and changes in global temperature.

Note that (unlike the code in #416) the current implementation computes the change in air temperature and then uses this change to compute the precipitation scaling. This means that precipitation changes are always consistent with temperature changes. Unlike #416, this code makes it impossible to set up a simulation in which elevation changes affect precipitation but **not** temperature. I suspect that this limitation should not be a problem.

@flocolleoni, @talbrecht : please let me know what you think.

**Comment by talbrecht** [comment]
I support option 2 (with the modified naming and independent of lapse rate and temperature related code), as it provides the means of disentangling different aspects of precipitation and temperature forcing.

**Comment by flocolleoni** [comment]
Dear Constantine,

Thank you for your very detailed input!
I agree with Torsten. I also support option 2 since then one can
activate/deactivate the different corrections.

**Comment by ckhroulev** [comment]
I just tagged the next PISM version (v1.2) which includes a way to adjust precipitation or SMB using an exponential factor using a change in temperature resulting from a lapse rate correction. Please see [Adjustments using modeled change in surface elevation](https://pism-docs.org/sphinx/climate_forcing/atmosphere.html#adjustments-using-modeled-change-in-surface-elevation) in the manual.

**Comment by talbrecht** [comment]
This is great. Thank you so much. I will test it soon. In the documentation we should distinguish between the two different lapse rates gamma_T and gamma_P.

**Comment by flocolleoni** [comment]
Dear Constantine,

Thank you very much for this work! I will test it ASAP.

Florence

**Comment by ckhroulev** [comment]
> In the documentation we should distinguish between the two different lapse rates gamma_T and gamma_P.

@talbrecht  OK.

**Comment by talbrecht** [comment]
When I am using the PIK temperature parameterization (`-atmosphere pik`, with annual mean and summer temperature inferred from multi regression fit), which includes already some temperature lapse rate (8.3K/km and 9.5K/km respectively), I cannot add another optional `-temp_lapse_rate`, which is however needed for the `-precip_adjustment scale`. In my older code I directly defined a `m_precip_exp_factor*m_temp_lapse_rate_scale=0.07*8.2=0.574` while the `m_temp_lapse_rate` was set to 0. What could be a simple solution here for the new PISM implementation?

```
-atmosphere pik,elevation_change -atmosphere_pik era_interim_lon \
          -atmosphere_pik_file $atmfile  \
          -atmosphere_lapse_rate_file $origfile \
          -temp_lapse_rate $tlapse -precip_adjustment scale \
          -surface pdd"
```
